.github/workflows/tests.yaml

```diff
@@ -0,0 +1,69 @@
+on: [push, pull_request]
+
+name: check & tests
+
+jobs:
+  check:
```

Comment:
This job is redundant because the `cargo check` is kind of embedded in the `cargo test` command which is run in the `test` job below.

Reply:
It would be useful to have a `fmt` job that checks the code formatting. You can use `cargo fmt -- --check` for that.

Reply:
You could use [this](https://github.com/lambdaclass/AES_zero_knowledge_proof_circuit/blob/master/.github/workflows/ci.yml) as an example.

Reply:
`cargo check` is a faster way to verify that doing `cargo build` or `cargo test`. So the intention is to have a failing job as soon as possible. 

From docs:
> This will essentially compile the packages without performing the final step of code generation, which is faster than running cargo build

If this sounds like not needed, I'm fine with removing it; but just want to make that point. Let me know if you still prefer to remove this step!

Reply:
I think that you're making a good point, let's keep this job!